runtime/aws/Amazon Elastic Container Registry/check-ecs-container-privileges.ts

```diff
@@ -0,0 +1,124 @@
+import {
+	ECSClient,
+	ListTaskDefinitionsCommand,
+	DescribeTaskDefinitionCommand
+} from "@aws-sdk/client-ecs";
+import { printSummary, generateSummary } from "../../utils/string-utils.js";
+import { ComplianceStatus, type ComplianceReport, type RuntimeTest } from "../../types.js";
+
+interface ContainerDefinition {
+	name?: string | undefined;
+	privileged?: boolean;
+}
+
+async function checkEcsContainerPrivileges(
+	region: string = "us-east-1"
+): Promise<ComplianceReport> {
+	const client = new ECSClient({ region });
+	const results: ComplianceReport = {
+		checks: []
+	};
+
+	try {
+		let nextToken: string | undefined;
+		let taskDefinitionsFound = false;
+
+		do {
+			// List all task definitions
+			const listCommand = new ListTaskDefinitionsCommand({
+				nextToken
+			});
+			const response = await client.send(listCommand);
+
+			if (!response.taskDefinitionArns || response.taskDefinitionArns.length === 0) {
+				if (!taskDefinitionsFound) {
+					results.checks = [
+						{
+							resourceName: "No Task Definitions",
+							status: ComplianceStatus.NOTAPPLICABLE,
+							message: "No ECS task definitions found in the region"
+						}
+					];
+					return results;
+				}
+				break;
+			}
+
+			taskDefinitionsFound = true;
+
+			// Check each task definition
+			for (const taskDefArn of response.taskDefinitionArns) {
```

Comment:
@cc-shekher AWS API limits only 100 task definitions in pagination. You should first get all task definitions and then apply further logic,

ex runtime/aws/Amazon CloudWatch/aws_cloudwatch_alarm_action_check.ts